flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/lineage/TableColumnLineageEdge.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.lineage;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.streaming.api.lineage.LineageDataset;
+
+import java.util.List;
+
+/**
+ * Column lineage from source table columns to each sink table column, one sink column may be
+ * aggregated by multiple tables and columns.
+ */
+@PublicEvolving
+public interface TableColumnLineageEdge {
+
+    /** The dataset for source dataset. */
+    LineageDataset source();
+
+    /**
+     * Columns from one source table of {@link org.apache.flink.streaming.api.lineage.LineageEdge}
+     * to the sink column. Each sink column may be computed from multiple columns from source, for
+     * example, avg operator from two columns in the source.
+     */
+    List<String> sourceColumns();
+
+    /* Sink table column. */
+    String sinkColumn();
```

Comment:
Can't we have 1 source column to multiple sink columns? I am wondering if this should be many to many?

Reply:
Yes, it is possible. If so, there will be multiple edges for the source column.